testnet/stacks-node/src/run_loop/neon.rs

```diff
@@ -167,6 +167,18 @@ impl RunLoop {
         );
 
         let pox_constants = burnchain.get_pox_constants();
+        let epochs = burnchain.get_stacks_epochs();
+        if !check_chainstate_db_versions(
+            &epochs,
+            &self.config.get_burn_db_file_path(),
+            &self.config.get_chainstate_path_str(),
+        )
+        .expect("FATAL: unable to query filesystem or databases for version information")
+        {
+            panic!(
+                "FATAL: chainstate database(s) are not compatible with the current system epoch"
+            );
+        }
```

Comment:
Have you run this branch on an empty chainstate directory on mainnet or testnet? When I made the `connect_dbs()` change in #2931, calling it before the `burnchain.start()` led to a panic, because the burnchain doesn't have a `first_block_hash` yet, so I had to move the `connect_dbs` invocation after `burnchain.start()`, and I wonder if the same problem would crop up here.

Reply:
+1 The change looks good to me, but does it work on a real node?

Reply:
Yup, it definitely works.